rtos/source/ThisThread.cpp

```diff
@@ -251,4 +251,32 @@ const char *get_name()
 #endif
 }
 
+osPriority_t ThisThread::get_priority()
+{
+#if MBED_CONF_RTOS_PRESENT
+    osThreadId_t id = osThreadGetId();
+    if (id == nullptr) {
+        return osPriorityError;
+    }
+
+    return static_cast<osPriority_t>(osThreadGetPriority(id));
+#else
+    return static_cast<osPriority_t>(osPriorityError);
+#endif
+}
+
+osStatus_t ThisThread::suspend()
+{
+#if MBED_CONF_RTOS_PRESENT
+    osThreadId_t id = osThreadGetId();
+    if (id == nullptr) {
+        return osError;
+    }
+
+    return static_cast<osStatus_t>(osThreadSuspend(id));
+#else
+    return static_cast<osStatus_t>(osError);
```

Comment:
Should we error here or do something more sensible for applications instead?

Reply:
What's the use case here? Why would you want to suspend the current thread? I would suggest adding it as `Thread::suspend()`, matching the actual CMSIS RTOS API.

There's no wake API to make it resume being added?

Are you sure you don't want a `ConditionVariable` or something?

What to do with bare metal depends how you're intending to wake it up.

Reply:
This function probably doesn't do what you want it to do. `osThreadSuspend` will suspend a thread until it is resumed using `osThreadResume` we are not providing it so basically we are switching current thread forever.
Why can't you use yield which is already defined?

Reply:
Yes, we want to suspend and then delete. Yield will not work because the schedule may begin running the thread we wanted to suspend before we've had a chance to delete it.